google-cloud-translate/lib/google/cloud/translate/service.rb

```diff
@@ -77,10 +76,118 @@ def inspect
 
         protected
 
+        def post path, body = nil
+          response = execute do
+            http.post path do |req|
+              req.headers.merge! default_http_headers
+              req.body = body unless body.nil?
+
+              if @key
+                req.params = { key: @key }
+              else
+                @credentials.sign_http_request req
+              end
+            end
+          end
+
+          return JSON.parse(response.body)["data"] if response.success?
+
+          fail Google::Cloud::Error.gapi_error_class_for(response.status)
```

Comment:
Can we extract error details from the response, and put them in the `Google::Cloud::Error`?